WordPress/Classes/System/SplitViewRootPresenter.swift

```diff
@@ -361,17 +389,50 @@ extension SplitViewRootPresenter: UISplitViewControllerDelegate {
             sitePickerPopoverVC?.presentingViewController?.dismiss(animated: true)
         }
     }
-}
 
-extension SplitViewRootPresenter: SiteMenuViewControllerDelegate {
-    func siteMenuViewController(_ siteMenuViewController: SiteMenuViewController, showDetailsViewController viewController: UIViewController) {
-        if viewController is UINavigationController ||
-            viewController is UISplitViewController {
-            splitVC.setViewController(viewController, for: .secondary)
-        } else {
-            // Reset previous navigation or split stack
-            let navigationVC = UINavigationController(rootViewController: viewController)
-            splitVC.setViewController(navigationVC, for: .secondary)
+    func splitViewController(_ svc: UISplitViewController, willShow column: UISplitViewController.Column) {
+        if column == .primary, let selection = displayingContent?.selection, selection != sidebarViewModel.selection {
+            sidebarViewModel.selection = selection
         }
     }
 }
+
+// MARK: - Content displayed within the split view, alongside the sidebar
+
+/// This protocol is an abstraction of the `supplimentary` and `secondary` columns in a split view.
+///
+/// When in full-screen mode, `SplitViewRootPresenter` presents a triple-column split view. The sidebar is displayed in
+/// the primary column, which is always accessible. The `supplimentary` and `secondary` columns display different
+/// content, depending on what users choose from the sidebar.
+protocol SplitViewDisplayable: AnyObject {
+    var supplimentary: UINavigationController { get }
+    var secondary: UINavigationController? { get set }
```

Comment:
`var secondary: UINavigationController?` should not be optional because it you have to display something in the `.secondary` tab, even if has to be a "Not Found" or "Something Went Wrong" or "No Selection" screen.

Reply:
**Note**: I think it's OK to hardcode secondary to `UINavigationController` because that's the default `UISplitViewController` behavior. By default, `UISplitViewController` retains the same `UINavigationController` when you change the `.secondary` column creating a navigation stack. I disabled this behavior by making `.secondary` column a navigation controller. It's unlikely that it can ever be anything else and you can always hide the navigation bar completely if that's what you want.